sig-and-wg/sig-core/proposals/two-phase-connector-service.md

```diff
@@ -0,0 +1,100 @@
+
+## Goal
+
+1. The Connector service handles client certificate provisioning for an App Registry connection.
+1. An endpoint for getting metadata information with App Registry URL, Event service URL is exposed and accessible with the use of the client certificate. Metadata can be extended to support automated communication between Kyma and connected applications. Example, the Event service endpoint can be changed, and application configuration must be updated.
+1. The Connector service handles client certificate provisioning for a Kyma runtime using the first client certificate.
+1. The new client certificate can be requested using the previous one (certificate rotation)
+1. The Connector service can trigger the creation of the application custom resource if it doesn't exist yet.
+
+## Suggested solution
+
+Connector service is extended to work with two phases:
+
+  - In the first phase, the Connector service is returning cluster metadata for CSR signing and information about App Registry URL.
+  - In the second phase, the Connector service is returning metadata of the Kyma runtime cluster for which a second client certificate can be obtained.
+
+Both phases can be realized using the same Connector Service, or there could be two separated Connector Service (central one handling App Registry for multiple Kyma runtimes and seconds one in each Kyma runtime)
+
+
+### Flow Diagram
+
+![Connector Service Flow](assets/connector-service-flow.svg)
+
+#### Application is requesting client certificate
+
+1. An application is requesting CSR information by sending registration token.
+2. Information about CSR requirement and URL for signing it is returned.
+3. CSR is created and send for a signing.
+4. The client certificate is returned with information about App Registry endpoint and Info endpoint.
+
+**Security:** The security is based on one-time tokens.
+
+#### Application is registering metadata
+
+1. An application is registering all APIs and Event catalogs in App Registry. It might be a single call or multiple separated calls.
+2. An application is able to update registered metadata.
+
+**Security:** The client certificate is representing the connected application.
+
+#### Application is getting Kyma metadata
+
+1. An application calls info endpoint.
+2. The information about Kyma runtime is returned. The information contains: App Registry URL, Event service URL, Connector service in Runtime (with connection token.)
+3. An application is updating its configuration if something changed (e.g., new Event Service URL, the certificate is expiring, and a new one must be fetched.)
+4. The calls should be executed periodically (pulling). It will be used as a heartbeat from the connected application.
+
+**Security:** The client certificate is representing the connected application.
+
+#### Application is connecting to the Kyma runtime
+
+1. An application is requesting CSR information by sending registration token returned in the first phase.
```

Comment:
Do you mean a connection token mentioned in the previous section (point 2)?
How the information about connection/registration token is trusted?

Reply:
Done